src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,506 @@
+ * stateful information (such as items added in the shopping cart in an online
+ * store) or to record the user's browsing activity (including clicking
+ * particular buttons, logging in, or recording which pages were visited in
+ * the past). They can also be used to remember arbitrary pieces of information
+ * that the user previously entered into form fields such as names, addresses,
+ * passwords, and credit card numbers.
+ *
+ * @link https://tools.ietf.org/html/rfc6265
+ * @link https://en.wikipedia.org/wiki/HTTP_cookie
+ */
+class Cookie implements CookieInterface
+{
+
+    use CookieCryptTrait;
+
+    /**
+     * Cookie name
+     *
+     * @var string
+     */
+    protected $name = '';
+
+    /**
+     * Raw Cookie value
+     *
+     * @var string|array
+     */
+    protected $value = '';
+
+    /**
+     * @var bool
+     */
+    protected $isExpanded = false;
+
+    /**
+     * Expiration time
+     *
+     * @var int
+     */
+    protected $expiresAt = 0;
+
+    /**
+     * Path
+     *
+     * @var string
+     */
+    protected $path = '';
+
+    /**
+     * Domain
+     *
+     * @var string
+     */
+    protected $domain = '';
+
+    /**
+     * Secure
+     *
+     * @var bool
+     */
+    protected $secure = false;
+
+    /**
+     * HTTP only
+     *
+     * @var bool
+     */
+    protected $httpOnly = false;
+
+    /**
+     * Constructor
+     *
+     * The constructors args are similar to the native php setcookie() method.
+     *
+     * @link http://php.net/manual/en/function.setcookie.php
+     * @param string $name Cookie name
+     * @param string|array $value Value of the cookie
+     * @param \DateTimeInterface|null $expiresAt Expiration time and date
+     * @param string $path Path
+     * @param string $domain Domain
+     * @param bool $secure Is secure
+     * @param bool $httpOnly HTTP Only
+     */
+    public function __construct($name, $value = '', $expiresAt = null, $path = '', $domain = '', $secure = false, $httpOnly = false)
+    {
+        $this->validateName($name);
+        $this->setName($name);
+        $this->setValue($value);
+        $this->setDomain($domain);
+        $this->setHttpOnly($httpOnly);
+        $this->setPath($path);
+
+        if ($expiresAt !== null) {
+            $this->expiresAt($expiresAt);
+        }
+    }
+
+    /**
+     * Builds the expiration value part of the header string
+     *
+     * @return string
+     */
+    protected function _buildExpirationValue()
+    {
+        return sprintf(
+            'expires=%s',
+            gmdate('D, d-M-Y H:i:s T', $this->expiresAt)
+        );
+    }
+
+    /**
+     * Returns a header value as string
+     *
+     * @return string
+     */
+    public function toHeaderValue()
+    {
+        $headerValue[] = sprintf('%s=%s', $this->name, urlencode($this->value));
+
+        if ($this->expiresAt !== 0) {
+            $headerValue[] = $this->_buildExpirationValue();
+        }
+        if (!empty($this->path)) {
```

Comment:
There's little probability that there would be `0` in path but still this could be `$this->path !== ''`